src/MICore/CommandFactories/MICommandFactory.cs

```diff
@@ -524,6 +524,16 @@ public virtual bool IsAsyncBreakSignal(Results results)
             return (results.TryFindString("reason") == "signal-received" && results.TryFindString("signal-name") == "SIGINT");
         }
 
+        /// <summary>
+        /// Determines if a new external console should be spawned on non-Windows platforms for the debugger+app
+        /// </summary>
+        /// <param name="localLaunchOptions">[required] local launch options</param>
+        /// <returns>True if an external console should be used</returns>
+        public virtual bool UseExternalConsoleForLocalLaunch(LocalLaunchOptions localLaunchOptions)
+        {
+            return localLaunchOptions.UseExternalConsole && String.IsNullOrEmpty(localLaunchOptions.MIDebuggerServerAddress) && !localLaunchOptions.IsCoreDump;
```

Comment:
Is the default on this true? gdb will deadlock for us if this isn't set. Eventually, we'd probably not want to throw the console on attach, but again, today gdb deadlocks.